test/e2e/tests/test_health_check.py

```diff
@@ -0,0 +1,201 @@
+
+"""Integration tests for the Route53 HealthCheck resource
+"""
+
+import pytest
+import random
+import socket
+import struct
+import time
+
+from acktest import tags
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import service_marker, get_route53_resource, create_route53_resource, delete_route53_resource, patch_route53_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.tests.helper import Route53Validator
+
+RESOURCE_PLURAL = "healthchecks"
+
+# Time to wait after modifying the CR for the status to change
+MODIFY_WAIT_AFTER_SECONDS = 10
+
+CREATE_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@pytest.fixture
+def health_check(request):
+    health_check_name = random_suffix_name("health-check", 32)
+    ip_address = socket.inet_ntoa(struct.pack('>I', random.randint(1, 0xffffffff)))
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["HEALTH_CHECK_NAME"] = health_check_name
+    replacements["IP_ADDR"] = ip_address
+    replacements["TAG_KEY"] = "initialtagkey"
+    replacements["TAG_VALUE"] = "initialtagvalue"
+
+    ref, cr = create_route53_resource(
+        "healthchecks",
+        health_check_name,
+        "health_check",
+        replacements,
+    )
+
+    yield ref, cr
+
+    delete_route53_resource(ref)
+
+
+def patch_health_check(ref):
+    ip_address = socket.inet_ntoa(struct.pack('>I', random.randint(1, 0xffffffff)))
+    updates = {
+        "spec": {
+            "healthCheckConfig": {
+                "iPAddress": ip_address,
+            }
+        }
+    }
+    patch_route53_resource(ref, updates)
+    return get_route53_resource(ref)
+
+
+@service_marker
+@pytest.mark.canary
+class TestHealthCheck:
+    @pytest.mark.resource_data({'tag_key': 'key', 'tag_value': 'value'})
+    def test_crud(self, route53_client, health_check):
+        ref, cr = health_check
+
+        health_check_id = cr["status"]["id"]
+
+        assert health_check_id
+
+        # Check health check exists in AWS
+        route53_validator = Route53Validator(route53_client)
+        route53_validator.assert_health_check(cr)
+
+        # Update health check resource and check that the value is propagated to AWS
+        updated = patch_health_check(ref)
+        assert updated["spec"]["healthCheckConfig"]["iPAddress"] != cr["spec"]["healthCheckConfig"]["iPAddress"]
+
+        # Check health check has been updated in AWS
+        route53_validator.assert_health_check(cr)
+
+        # Delete k8s resource
+        _, deleted = k8s.delete_custom_resource(ref)
+        assert deleted is True
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        # Check health check no longer exists in AWS
+        route53_validator.assert_health_check(cr, exists=False)
+
+
+    # @pytest.mark.resource_data({'tag_key': 'initialtagkey', 'tag_value': 'initialtagvalue'})
+    # def test_crud_tags(self, route53_client, health_check):
```

Comment:
I see that we have hooks for the `syncTags` should we uncomment this now?

Reply:
Well the issue is is that it does not work. The syncTags function works fine, But it only updates the tags when there are also changes in the other spec. So I think there is an issue with the delta function that does not see differences when we only edit the Tags in the spec. I don't know if I am missing something or if there is an issue with the default delta function.

Reply:
Ok i took a look at how it was done on the HostedZone resource, and got it done on the HealthCheck one. I uncommented the test.